### ISSUE TITLE
Providing enough information for rust-battery / starship

### ISSUE THREAD

**Issue by alexanderadam** [termux/termux-api#364]
**Feature description**

The Rust library [`rust-battery`](https://github.com/svartalf/rust-battery) allows Rust applications, like the popular prompt [starship](https://github.com/starship/starship), to consume battery information (independent from the underlying operating system). The current output isn't enough and is [leading to compile errors](https://github.com/svartalf/rust-battery/issues/69).

[The maintainer](https://github.com/svartalf/rust-battery/issues/33#issuecomment-671069623) @svartal [wrote](https://github.com/svartalf/rust-battery/issues/33#issuecomment-671069623):

> I'm definitely up for adding functionality needed to run `battery` (and `starship` as a consequence) inside of the `termux` env, but it is out of my scope to work on that directly; if you want to raise an issue for `termux-api` and communicate with its developers to extend provided API, that would be great.

The current output of `termux-battery-status` is
```json
{
  "health": "GOOD",
  "percentage": 59,
  "plugged": "UNPLUGGED",
  "status": "DISCHARGING",
  "temperature": 27.200000762939453,
  "current": -693
}
```
but [the current `Battery` struct](https://docs.rs/battery/0.7.5/battery/struct.Battery.html) needs:

| Attribute | Description |
| --- | --- |
| `state_of_charge` | Battery state of charge |
| `energy` | Amount of energy currently available in the battery |
| `energy_full` | Amount of energy in the battery when it's considered full. |
| `energy_full_design` | Amount of energy the battery is designed to hold when it's considered full. |
| `energy_rate` | Amount of energy being drained from the battery. |
| `voltage` | Battery voltage. |
| `state_of_health` | Gets battery state of health. |
| `state` | Battery current state. |
| `technology` | Battery technology. |
| `temperature` | Battery temperature. |
| `cycle_count` | Number of charge/discharge cycles. |
| `vendor` | Battery vendor. |
| `model` | Battery model. |
| `serial_number` | Battery serial number. |
| `time_to_full` | Remaining time till full battery. |
| `time_to_empty` | Remaining time till empty battery. |

Would it be possible to extend `termux-battery-status` to enable support for `rust-battery`?

**PS:** Thank you so much for your work! Termux is awesome and enables an impressive level of productivity on Android! :pray:

**Comment by Grimler91** [comment]
Since we are accessing the android API we can only get the values defined in it, these are listed here: https://developer.android.com/reference/android/os/BatteryManager. 

> but the current Battery struct needs:

Does "needs" mean that these values are absolutely necessary or can some be set to `NULL`?

`technology`, `vendor` and `model` would have to be implemented manually as a huge map for every phone and I do not think anyone wants to implement that (and it would not be 100 % correct anyways as many (older) phones have replaceable batteries). 

`serial_number` would be impossible to access.

How does other OS's handle this, are `technology`, `vendor`, `model` and `serial_number` always accessible?

Since rust-battery would anyways need to handle termux/android as a special OS I think it would make sense to set the unknown fields to `NULL` there rather than having termux-api return some fields which are always `NULL`.

**Comment by xeffyr** [comment]
> Since we are accessing the android API we can only get the values defined in it, these are listed here: https://developer.android.com/reference/android/os/BatteryManager.

Yes, because other values are vendor specific and usually exposed through sysfs only at
```
/sys/class/power_supply/battery
```
Entries here are device-specific, some of them may be absent or have access restricted by SELinux.

> How does other OS's handle this, are technology, vendor, model and serial_number always accessible?

On laptop these values accessible through sysfs:
```
~ $ ls /sys/class/power_supply/BAT0
alarm                   device              model_name     subsystem
capacity                energy_full         power          technology
capacity_level          energy_full_design  power_now      type
charge_start_threshold  energy_now          present        uevent
charge_stop_threshold   hwmon2              serial_number  voltage_min_design
cycle_count             manufacturer        status         voltage_now
```
Unlike Android devices, entries here are standard.

**Comment by alexanderadam** [comment]
Sorry, [the maintainer of `rust-battery` just wrote](https://github.com/svartalf/rust-battery/issues/33#issuecomment-689370864)

> to this point I don't see how Android' `BatteryManager` data can be mapped to `Battery` from this crate.
